client/src/App.js

```diff
@@ -3,32 +3,34 @@ import { MuiThemeProvider } from '@material-ui/core';
 import { BrowserRouter, Route, Switch } from 'react-router-dom';
 
 import { theme } from './themes/theme';
+import './App.css';
 
-import LogIn from './pages/Login';
+import Home from './pages/Home';
 import SignUp from './pages/SignUp.js';
 import Profile from './pages/Profile.js';
 import NotFound from './pages/NotFound.js';
+import PostPage from './pages/Post/PostPage';
 
+import Login from './components/Dialog/Login/Login';
 import PostDialog from './components/Dialog/PostDialog/PostDialog';
 import BoardDialog from './components/Dialog/BoardDialog/BoardDialog';
 import InterestQuizDialog from './components/Dialog/InterestQuizDialog/QuizDialog';
-import './App.css';
-import PostPage from './pages/Post/PostPage';
 
 function App () {
     return (
         <MuiThemeProvider theme={theme}>
             <BrowserRouter>
                 <Switch>
-                    <Route exact path='/' component={LogIn} />
-                    <Route exact path='/signup' component={SignUp} />
+                    <Route exact path='/' component={Home} />
+                    <Route path='/post/:id' component={PostPage} />
+                    <Route path='/profile/:username' component={Profile} />
+                    <Route exact path='/' component={NotFound} />
```

Comment:
I don't think this ever gets rendered

Reply:
just remove the exact from the NotFound Route?

Reply:
Yes, but also make sure the routes below it (outside the switch) get moved up, otherwise NotFound will be rendered together with them

Reply:
For routes like:
  <Route path="/profile/:username/post/create" component={PostDialog} />
   <Route path="/profile/:username/board/create" component={BoardDialog} />
You can actually moved them inside the actual Profile component. It's fine to nest routes

Reply:
ah, ok. That should make things simpler.

Reply:
So, I nested the profile routes and they work. If I wanted to nest the login route in the home component and have it appear on load. 

`componentDidMount () {
        this.props.history.push('/login');
    }`

I moved the login route to the home component `<Route path={this.props.match.url + '/login'} component={Login} />`
but its not working ...

Reply:
Is it forming the route as "//login" because this.props.match.url= "/" ?

Reply:
If not, can you push your changes so I can look at the new code

Reply:
ok

Reply:
I think i found a solution, I removed the exact on the route from app.js

Reply:
Awesome